server/get_header.go

```diff
@@ -59,218 +70,384 @@ func (m *BoostService) getHeader(log *logrus.Entry, slot phase0.Slot, pubkey, pa
-			if err != nil {
-				log.WithError(err).Warn("error parsing response content type")
-				respContentType = MediaTypeJSON
-			}
-			log = log.WithField("respContentType", respContentType)
+	// process the bids and select the one with the best value
+	for _, rb := range relayBids {
+		m.processBid(log, rb.relay, rb.bid, rb.contentType, parentHashHex, &result, relays, slot)
+	}
 
-			// Get the optional version, used with SSZ decoding
-			respEthConsensusVersion := resp.Header.Get(HeaderEthConsensusVersion)
-			log = log.WithField("respEthConsensusVersion", respEthConsensusVersion)
+	// Set the winning relays before returning
+	result.relays = relays[BlockHashHex(result.bidInfo.blockHash.String())]
 
-			// Decode bid
-			bid := new(builderSpec.VersionedSignedBuilderBid)
-			err = decodeBid(respBytes, respContentType, respEthConsensusVersion, bid)
-			if err != nil {
-				log.WithError(err).Warn("error decoding bid")
-				return
-			}
+	if len(result.relays) > 0 {
+		RecordWinningBidValue(result.bidInfo.value.Float64())
+	}
 
-			// Skip if bid is empty
-			if bid.IsEmpty() {
-				log.Debug("skipping empty bid")
-				return
-			}
+	return result, nil
+}
 
-			// Getting the bid info will check if there are missing fields in the response
-			bidInfo, err := parseBidInfo(bid)
-			if err != nil {
-				log.WithError(err).Warn("error parsing bid info")
-				return
-			}
+// handleTimingGamesGetHeader implements timing games strategy for a relay
+// Returns the latest bid and its content type from multiple timed requests
+func (m *BoostService) handleTimingGamesGetHeader(
+	log *logrus.Entry,
+	relayConfig types.RelayConfig,
+	url string,
+	slotUID uuid.UUID,
+	userAgent string,
+	proposerAcceptContentTypes string,
+	msIntoSlot uint64,
+	timeoutLeftMs uint64,
+) (*builderSpec.VersionedSignedBuilderBid, string) {
+	relay := relayConfig.RelayEntry
+
+	// wait til target time is configured
+	if relayConfig.TargetFirstRequestMs > 0 {
+		delayMs := relayConfig.TargetFirstRequestMs - msIntoSlot
+		if delayMs > 0 {
+			log.WithFields(logrus.Fields{
+				"targetMs":   relayConfig.TargetFirstRequestMs,
+				"msIntoSlot": msIntoSlot,
+				"delayMs":    delayMs,
+			}).Debug("waiting to send header request via timing games")
+			timeoutLeftMs -= delayMs
+			time.Sleep(time.Duration(delayMs) * time.Millisecond)
+		}
+	}
 
-			// Ignore bids with an empty block
-			if bidInfo.blockHash == nilHash {
-				log.Warn("relay responded with empty block hash")
-				return
-			}
+	// send multiple requests at frequency intervals
+	if relayConfig.FrequencyGetHeaderMs > 0 { //nolint:nestif
+		log.WithFields(logrus.Fields{
+			"frequencyMs":   relayConfig.FrequencyGetHeaderMs,
+			"timeoutLeftMs": timeoutLeftMs,
+		}).Debug("sending multiple header requests via timing games")
+
+		var bidResults []bidResult
+		var mu sync.Mutex
+
+		// keep sending requests until time runs out
+		var wg sync.WaitGroup
+		for timeoutLeftMs > 0 {
```

Comment:
Did we consider using `time.Ticker` with `select` inside this `for` to launch `go func()`? We could do something like:
```go
timeoutCh := time.After(timeoutLeftMs)
ticker := time.NewTicker(time.Duration(relayConfig.FrequencyGetHeaderMs) * time.Millisecond)
for {
	select {
		case <-ticker.C:
			// go func()
		case <-timeoutCh:
			break
	}
}
```